src/smtml/expr.ml

```diff
@@ -382,6 +384,10 @@ let rec binop ty op hte1 hte2 =
     when Bitvector.eqz bv ->
     hte1
   | (Add | Or), _, Val (Bitv bv) when Bitvector.eqz bv -> hte1
+  | Add, _, Val (Real 0.) -> hte1
+  | Add, Val (Real 0.), _ -> hte2
+  | Add, _, Val (Num (F32 0l)) -> hte1
+  | Add, Val (Num (F32 0l)), _ -> hte2
```

Comment:
I don't think this is right. On floats `+0. + -0.` is going to give `+0.`. (I.e. the neutral element is `-0` and not `+0`.)